Add CONTRIBUTING.md for website launch (#172)
Naoray/scribe: CONTRIBUTING.md

5. **Run the checks.** Before pushing:
   - `go build ./...`
   - `go test ./...`
   - `gofmt -w` on any files you touched (CI fails on unformatted code).
6. **Write a useful commit history.** Short imperative subjects, one logical change per commit. Examples from `git log`: `fix: preserve tools on list TUI updates`, `perf: lazy-init command dependencies`. Avoid mixed refactors in a single commit.
7. **Open the PR.** Describe the user-visible change, note any migration impact or risk, and include terminal output or a screenshot when changing TUI behavior. Link the issue you are closing with `Closes #N`.
8. **Respond to review.** Push fix-up commits to the same branch — do not force-push during review unless asked. The maintainer will squash-merge to keep `main` clean.

For larger changes (new commands, new envelope fields, breaking flag changes), draft a short design note in the PR description before writing code. A discussion of the public contract is cheaper than rewriting an implementation.

## Local development

### Prerequisites

> - Go 1.22 or newer (`go version`).

Author: `go.mod` declares `go 1.26.1`. Building with 1.22 will fail.

```suggestion
- Go 1.26 or newer (`go version`).
```

## Adding a registry

A registry is a GitHub repository that ships scribe-compatible skills declared in a top-level `scribe.yaml` manifest. Anyone can host one — your team's private skills repo, a community pack, or a personal collection.

### Create a new registry

The fastest path is the scaffold command:

```bash
scribe registry create
```

It walks you through creating a GitHub repository (private or public), seeding `scribe.yaml`, and pushing the initial commit. Pass flags to skip prompts in scripts:

```bash
scribe registry create --owner my-org --repo my-skills --private --json
```

If you prefer to bootstrap by hand, the manifest looks like this:

```yaml
# scribe.yaml
version: 1
skills:
  - name: review-checklist
    path: skills/review-checklist
    description: Apply the team's review checklist before opening a PR.
```

Each skill lives in its own directory with a `SKILL.md` following the [agentskills.io](https://agentskills.io) format. Anything that works with `skills.sh` or Paks works with scribe.

### Connect to an existing registry

```bash
scribe registry connect my-org/my-skills
scribe sync
```

> `registry connect` adds the repo to `~/.scribe/registries.yaml` and fetches its catalog. `sync` then projects the registry's skills into the agents you have detected.

Author: There is no `~/.scribe/registries.yaml`. Per `internal/paths/paths.go`, the only files are `config.toml`, `config.yaml`, and `state.json`. Connected registries live inside `config.{yaml,toml}`.

```suggestion
`registry connect` adds the repo to `~/.scribe/config.yaml` and fetches its catalog. `sync` then projects the registry's skills into the agents you have detected.
```

For one-off discovery without committing:

```bash
scribe browse --registry my-org/my-skills
```

### Share a local skill into a registry

```bash
scribe registry add review-checklist
```

> This opens a PR against the registry repo with the skill content. You need write access (or fork-and-PR rights) to the registry. See `docs/commands.md` for the full registry command surface — `enable`, `disable`, `forget`, `resync`, `list`, `migrate`.

Author: `scribe registry add` does not open a PR — `internal/add/add.go:249` (and `:315` for package refs) calls `client.PushFiles(ctx, owner, repo, ...)` which commits straight to `main`. So "fork-and-PR rights" is misleading; either you have push access or you target a fork via `--registry your-fork/repo`.

```suggestion
This commits the skill files (and the manifest entry) directly to the registry's `main` branch. You need push access to the target registry; for public registries you don't own, fork it first and target your fork via `--registry owner/repo`. See `docs/commands.md` for the full registry command surface — `enable`, `disable`, `forget`, `resync`, `list`, `migrate`.
```
